python/ray/serve/_private/proxy_router.py

```diff
@@ -70,6 +70,9 @@ def update_routes(self, endpoints: Dict[DeploymentID, EndpointInfo]) -> None:
                     _source=DeploymentHandleSource.PROXY,
                 )
                 handle._set_request_protocol(self._protocol)
+                # Eagerly instantiate the router for each handle so it can receive
+                # the replica set from the controller.
+                handle._get_or_create_router()
```

Comment:
nit: would prefer that we add an argument to `get_handle` instead of adding a private method call here

Reply:
added `_lazy_router_initialization` to `get_deployment_handle` and `DeploymentHandle` constructor. This is set to false in the proxy.

Reply:
@edoakes Removed the parameter for now because it causes the router to be initialized before options passed in `.options()` are picked up by the initialized router, and there is no super clean way of adding the param and fixing that. Reverted to manually calling private method, will think more about how to add an official parameter for this.